frame/contracts/src/benchmarking.rs

```diff
@@ -21,91 +21,411 @@
 
 use crate::*;
 use crate::Module as Contracts;
+use crate::exec::StorageKey;
+use crate::schedule::API_BENCHMARK_BATCH_SIZE;
 
-use frame_benchmarking::{benchmarks, account};
+use frame_benchmarking::{benchmarks, account, whitelisted_caller};
 use frame_system::{Module as System, RawOrigin};
-use parity_wasm::elements::FuncBody;
-use sp_runtime::traits::Hash;
+use parity_wasm::elements::{Instruction, Instructions, FuncBody, ValueType, BlockType};
+use sp_runtime::traits::{Hash, Bounded, SaturatedConversion, CheckedDiv};
+use sp_std::{default::Default, convert::{TryFrom, TryInto}};
 
-macro_rules! load_module {
-	($name:expr) => {{
-		let code = include_bytes!(concat!("../fixtures/benchmarks/", $name, ".wat"));
-		compile_module::<T>(code)
-	}};
+/// How many batches we do per API benchmark. 
+const API_BENCHMARK_BATCHES: u32 = 20;
+
+#[derive(Clone)]
+struct WasmModule<T:Trait> {
```

Comment:
I find this code really hard to approach. Can you please sprinkle some comments here and there? Not necessarily in this PR though.

Reply:
I agree. I refactored all the helper functions and put the code into its own module. I even added comments. I think it is now way easier to understand.